Dojo 1.13.1 breaks FormData submission (from 1.13.0)

In ledgersmb/LedgerSMB#3799 a problem is being reported due to our project upgrading the included (built) Dojo from 1.13.0 to 1.13.1. The problem lies in code which - when flattened and stripped - looks like this:

```js
                     options["data"] = new FormData(this.domNode);
                     // FF doesn't add the clicked button
                    options["data"].append('action',
                                         this.clickedAction);
                      return xhr(url, options).then(
                          function(doc){
                              self.hide_main_div();
                              self.set_main_div(doc);
                          },
                          function(err){
                              self.show_main_div();
                              self.report_request_error(err);
                          });
```

where `xhr` is required as `dojo/request/xhr`.

Instead of posting the content of 'FormData`, the actual content being posted is `[object Object]`.

this is still an issue in dojo 1.14.2
I'm using

```
dojo.xhrPost({
		preventCache:true,
		handleAs: "json",
		form: f,
		url: "...",
		load: function(data) { 
			///do something
		}
	});
```
(sorry for using the old way - but this is legacy code)

that gives me a 'request canceled' error because of deepCopy running into a endless-recursion.
the reason seems to be, that the form contains  a <input type="checkbox"/> which has a 'form' property, pointing back to the form (being HTMLFormElement - where the checkbox is again contained etc.)

> RangeError: Maximum call stack size exceeded
>     at Boolean.toString (<anonymous>)
>     at isFunction (dojo.js.uncompressed.js:115)
>     at req.has (dojo.js.uncompressed.js:171)
>     at Object.deepCopy (dojo.js.uncompressed.js:11043)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
>     at Object.deepCopy (dojo.js.uncompressed.js:11052)
> dojo.js.uncompressed.js:11904 Error: request cancelled
>     at dojo.js.uncompressed.js:11829
>     at dojo.Deferred.promise.cancel.cancel (dojo.js.uncompressed.js:4181)
>     at Object.xhr (dojo.js.uncompressed.js:12084)
>     at Object.dojo.rawXhrPost.dojo.xhrPost (dojo.js.uncompressed.js:12117)

--- a/request/util.js
+++ b/request/util.js
@@ -10,12 +10,19 @@ define([
 	'../has'
 ], function(exports, RequestError, CancelError, Deferred, ioQuery, array, lang, Promise, has){
 
+	var HAS_FORM_ELEMENT = !!window.HTMLFormElement;
+	
 	function isArrayBuffer(value) {
 		return has('native-arraybuffer') && value instanceof ArrayBuffer
 	}
 
 	function isBlob(value) {
 		return has('native-blob') && value instanceof Blob
+	}
+	
+	function isFormElement(value) {
+		return HAS_FORM_ELEMENT && value instanceof HTMLFormElement //all other
+				|| !HAS_FORM_ELEMENT && value.tagName === "FORM"; //IE<=7
 	}
 
 	function isFormData(value) {
@@ -26,6 +33,7 @@ define([
 		return value &&
 			typeof value === 'object' &&
 			!isFormData(value) &&
+			!isFormElement(value) &&
 			!isBlob(value) &&
 			!isArrayBuffer(value)
 	}

--- a/request/util.js
+++ b/request/util.js
@@ -10,8 +10,6 @@ define([
 	'../has'
 ], function(exports, RequestError, CancelError, Deferred, ioQuery, array, lang, Promise, has){
 
-	var HAS_FORM_ELEMENT = !!window.HTMLFormElement;
-	
 	function isArrayBuffer(value) {
 		return has('native-arraybuffer') && value instanceof ArrayBuffer
 	}
@@ -21,8 +19,11 @@ define([
 	}
 	
 	function isFormElement(value) {
-		return HAS_FORM_ELEMENT && value instanceof HTMLFormElement //all other
-				|| !HAS_FORM_ELEMENT && value.tagName === "FORM"; //IE<=7
+		if(typeof HTMLFormElement !== 'undefined') { //all other
+			return value instanceof HTMLFormElement;
+		} else { //IE<=7
+			value.tagName === "FORM"
+		}
 	}
 
 	function isFormData(value) {

